Fix [SETUP-BUG] #1748

sifanLV opened the pull request:
## What changes are proposed in this pull request?
 
This patch fixes a setup issue with creating new MLflow instance with MS
SQL server.
 
## How is this patch tested?
 
Tested with MS Azure. Deployed as Docker image using Azure Kubernetes Services with Azure SQL database.
 
## Release Notes
 
Support using MS SQL Server database when creating new instances of MLflow. Setting that enables adding a `experiment_id = 0` for auto increment column.
 
### What component(s) does this PR affect?
 
- [ ] Serving

### How should the PR be classified in the release notes? Choose one:
 
- [X] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes

sifanLV committed:
Fix [SETUP-BUG] #1748

This patch fixes a setup issue with creating new MLflow instance with MS
SQL server. Tested with MS Azure.

sifanLV commented:
Hi @mparkhe,

I added this code to my installation because I could not start MLflow with an empty MSSQL database. 

From my experience in running MLflow with MSSQL in Azure, it seems that 'zero' default value cannot be enabled without this setting. In that sense, I changed the settings for MSSQL in this file solely for that purpose. With that being said, I do want to follow the usage structure of MLflow. I would be grateful if you could point me towards the right place to add settings for MSSQL within MLflow. 

Thanks.

mparkhe reviewed (commented):
Nuanced difference in behavior for databases, depending on default settings. This is a good change--recommendation to make slight changes.

Also could you add a unit test for this?

sifanLV commented on mlflow/store/sqlalchemy_store.py:
```diff
@@ -162,11 +162,15 @@ def make_managed_session():
     def _set_no_auto_for_zero_values(self, session):
```
Thank you for your explanation and instructions. I'll update this pull request accordingly.

sifanLV committed:
Update Fix [SETUP-BUG] #1748

Rename two methods for configuring MYSQL and MSSQL.
Add unit tests for updated methods.

sifanLV committed:
Change Method Names

sifanLV committed:
Fix Lint Issues

mparkhe reviewed (approved):
Thank you for making the changes with very informative comments. Also for adding the unit test support. Super valuable!
